airbyte_cdk/sources/declarative/resolvers/config_components_resolver.py

```diff
@@ -130,7 +172,21 @@ def resolve_components(
                 )
 
                 path = [path.eval(self.config, **kwargs) for path in resolved_component.field_path]
+                parsed_value = self._parse_yaml_if_possible(value)
+                updated = dpath.set(updated_config, path, parsed_value)
 
-                dpath.set(updated_config, path, value)
+                if parsed_value and not updated and resolved_component.create_or_update:
+                    dpath.new(updated_config, path, parsed_value)
 
             yield updated_config
+
+    @staticmethod
+    def _parse_yaml_if_possible(value: Any) -> Any:
+        if isinstance(value, str):
+            try:
+                import yaml
+
+                return yaml.safe_load(value)
+            except Exception:
+                return value
```

Comment:
When is it not the case that this isn't parsable in yaml? It seems like in most scenario this should always be parsable and if it wasn't we throw an error.

Would it be because the spec/config might be JSON instead of yaml?

Or at best we may want to have stricter error handling against a specific error type instead of all.